Use "Expand all" button to expand accordions

bkjohnson opened the pull request:
## Description

`cy.expandAccordions` was considerably changed in https://github.com/department-of-veterans-affairs/vets-website/pull/20051. A [support request](https://dsva.slack.com/archives/CBU0KDSB1/p1647609046050589) came up showing that `cy.expandAccordions` wasn't working since it wasn't finding the button (this may be related to a cypress update, not necessarily related to the changes in #20051 since that was working previously). Fortunately, the accordion component [has a new "Expand all" button](https://github.com/department-of-veterans-affairs/component-library/releases/tag/v7.4.2) which could improve the `cy.expandAccordions` implementation, and it also looks like it would resolve the support issue (see screenshot).

Unfortunately, that feature was removed from `vets-website` pending a [fix to facilities templates](https://github.com/department-of-veterans-affairs/va.gov-team/issues/38741) ([slack thread](https://dsva.slack.com/archives/C02730UEZPS/p1647469887046229)).

Once the "Expand all" feature for accordions is back in `vets-website`, this PR may resolve the issue with cypress not expanding the accordions.

## Original issue(s)
N/A

## Testing done

- Local cypress testing

## Screenshots

![image](https://user-images.githubusercontent.com/2008881/159041901-62c4a553-3913-4587-a3a4-1f55ceeedb89.png)

Gracefully handling `open-single` accordions:

![image](https://user-images.githubusercontent.com/2008881/159047605-4f8a4708-7533-430c-bf6d-124e68859d56.png)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

bkjohnson committed:
Use "Expand all" button to expand accordions

bkjohnson commented on src/platform/testing/e2e/cypress/support/commands/expandAccordions.js:
```diff
@@ -7,20 +7,12 @@ Cypress.Commands.add('expandAccordions', () => {
     // Check if va-accordion-item Web Component exists
-    if ($main.find('va-accordion-item').length > 0) {
-      cy.get('va-accordion-item')
+    if ($main.find('va-accordion').length > 0) {
+      cy.get('va-accordion')
```
@pdavies88 can you provide some more context here? Would we need to keep the `$main.find()` _and_ the `cy.get()`?

pdavies88 replied:
@bkjohnson Cypress isn't smart enough to combine conditionals so you have to wrap them instead and the if else conditionals are checked via determining if there is length in the find. So find determines the presence of the element and then the get finds the element. If you don't do the find and just the get and the element doesn't exist on the page being tested with the suite will error and fail. So find ensures that situation does not occur by determining its presence first.

bkjohnson committed:
Remove outer conditional

If expandAccordions is being called, there should be a va-accordion on
the page.

pdavies88 reviewed (approved):
In theory once we add back in the Expand All button this should pass for all the tests like the image you added to the PR but right now with us pulling that code out the test suites will fail.

bkjohnson committed:
Leave open-single accordions alone

bkjohnson commented on src/platform/testing/e2e/cypress/support/commands/expandAccordions.js:
```diff
@@ -6,23 +6,14 @@ Cypress.Commands.add('expandAccordions', () => {
-          }
-        });
-    }
+        .find('button[aria-label="Expand all accordions"]')
```
This implementation would throw an error if `expandAccordions` was called after the accordion was already expanded:

![image](https://user-images.githubusercontent.com/2008881/159049268-16726027-4ea4-431a-89c4-1cfe470a97ad.png)

Maybe this can be expected behavior.

pdavies88 replied:
@bkjohnson yeah so the aria-label changes from `Expand all accordions` to `Collapse all accordions` on click which is why that error is occurring. We could change the aria-label to `Expand or Collapse all accordions` in the Web Component and that would solve the issue if you'd like

bkjohnson committed:
Simplify selector

Teams shouldn't be setting `open-single="false"`

bkjohnson committed:
Use quotes in aria-expanded selector

This seems to be the thing that's causing the issue of the button not
being found

bkjohnson committed:
Revert "Expand all"-related changes

This can be added later

bkjohnson committed:
Add quotes to accordion item button selector

bkjohnson committed:
Reapply "Expand all" changes

Adding the quotes to "false" for the va-accordion-item button didn't
work like it did for the button outside of the shadow DOM

bkjohnson committed:
Remove quotes around aria-expanded selector

This doesn't actually seem necessary

bkjohnson committed:
Merge branch 'main' into improve-expandAccordions
